src/Data/Primitive/Vec.hs

```diff
@@ -83,6 +94,37 @@ import GHC.Word
 --
 data Vec (n :: Nat) a = Vec ByteArray#
 
+class Vectoring vector a | vector -> a where
+    type IndexType vector :: Data.Kind.Type
```

Comment:
What's the purpose of `IndexType`?

Reply:
It was to allow to have the same function when working in Exp  and not Exp. Given that the simplicity of converting non `Acc` code to `Acc` code is one of the main selling points of Accelerate, that sounded like a nice idea.